### PR TITLE
Requiring authentication for all API calls

### PR DESCRIPTION
**Pull request by clintonb**
- All API calls require authentication
- Added support for OAuth2 authentication
- All API endpoints default to using Django model permissions

ECOM-3895

### CODE REVIEW

**Comment by clintonb** [comment]
@bderusha @peter-fogg please review

**Comment by bderusha** [comment]
LGTM :+1:

File: course_discovery/apps/courses/tests/test_models.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def assert_course_attrs(self, course, attrs):
         for attr, value in attrs.items():
             self.assertEqual(getattr(course, attr), value)
 
-    @responses.activate  # pylint: disable=no-member
+    @responses.activate
```
Comment (peter-fogg) [inline course_discovery/apps/courses/tests/test_models.py:31]: For my own edification with respect to Pylint's occasionally-mysterious behavior, do you have any idea why the `disable` was necessary before?

File: pylintrc_tweaks [hunk 2]
```diff
@@ -8,4 +8,4 @@ const-rgx = (([A-Z_][A-Z0-9_]*)|(__.*__)|log|urlpatterns|logger|User)$
 DISABLE+= ,invalid-name,missing-docstring
 
 [TYPECHECK]
-ignored-classes+= ,WSGIRequest,UserFactory,CatalogFactory
+ignored-classes+= ,WSGIRequest,UserFactory,CatalogFactory,responses
```
Comment (peter-fogg) [inline pylintrc_tweaks:11]: Ah, I see.

**Comment by peter-fogg** [comment]
Send it to the front :+1:

File: pylintrc_tweaks (hunk 2 above)
Comment (clintonb) [inline pylintrc_tweaks:11]: More details: https://github.com/getsentry/responses/issues/74

File: requirements/test.txt [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Packages required for testing
```
Comment (bderusha) [inline requirements/test.txt:1]: :rocket: good whitespace cleanup! YOU ROCK(et)!

**Pull request merged**